client/client.go

```diff
@@ -334,15 +334,15 @@ func (cli *Client) NegotiateAPIVersion(ctx context.Context) {
 // we are connected with an old daemon without API version negotiation support,
 // and downgrade to the latest version of the API before version negotiation was
 // added (1.24).
-func (cli *Client) NegotiateAPIVersionPing(pingResponse types.Ping) {
+func (cli *Client) NegotiateAPIVersionPing(ctx context.Context, pingResponse types.Ping) {
 	if !cli.manualOverride {
-		cli.negotiateAPIVersionPing(pingResponse)
+		cli.negotiateAPIVersionPing(ctx, pingResponse)
 	}
 }
 
 // negotiateAPIVersionPing queries the API and updates the version to match the
 // API version from the ping response.
-func (cli *Client) negotiateAPIVersionPing(pingResponse types.Ping) {
+func (cli *Client) negotiateAPIVersionPing(_ context.Context, pingResponse types.Ping) {
```

Comment:
Same here; at least I would not pre-emptively change the signatures; more so for non-exported functions (we can change those at any time we need the context?)